core/src/core/usb.zig

```diff
@@ -513,7 +513,7 @@ pub const Dir = enum(u8) {
 /// Describes an endpoint within an interface
 pub const EndpointDescriptor = extern struct {
     /// Length of this struct, must be 7.
-    length: u8,
+    length: u8 = 7,
```

Comment:
I guess this could be replaced with a `@sizeOf(@This())`, i'd probably create a helper function like `sizeOfCheck`:

```zig
fn sizeOfCheck(comptime T: type, comptime expected: comptime_int) comptime_int {
    const actual = @sizeOf(T);
    if(actual != expected) {
        @compileError(std.fmt.comptimePrint("{s} was expected to be {d} bytes large, but actually has {d} bytes size.", .{
            @typeName(T), expected, actual,
        });
    }
    return actual;
}
```

This way, you hardcode the size of the field, but you also validate that it is actually `7`.

Reply:
This would be possible indeed, thought about that too.

Reply:
I've decided to remove the length fields altogether, as they're only used for calculating the length of the descriptors and the calculation is indeed wrong. An extern fields are padded and therefore do not always result in the correct size returned by sizeOf. For e.g. the example EndpointDescriptor is reported as 10 bytes even though the acutal size is 9. I've updated the examples to show how the _magicnumber_ is calculated and we should strive to hide the manual calculation from the user. Will do this in a followup PR.